src/libsyntax/lib.rs

```diff
@@ -14,6 +14,8 @@
 //!
 //! This API is completely unstable and subject to change.
 
+#![deny(bare_trait_objects)]
```

Comment:
For future reference: attributes like these go below the 4 lines of `#![doc]` below.